arch/x86/include/asm/barrier.h

```diff
@@ -19,9 +21,27 @@
 #define rmb() alternative("lock; addl $0,0(%%esp)", "lfence", X86_FEATURE_XMM2)
 #define wmb() alternative("lock; addl $0,0(%%esp)", "sfence", X86_FEATURE_XMM)
 #else
+#ifndef CONFIG_KTSAN
 #define mb() 	asm volatile("mfence":::"memory")
 #define rmb()	asm volatile("lfence":::"memory")
 #define wmb()	asm volatile("sfence" ::: "memory")
+#else /* CONFIG_KTSAN */
+#define mb()								\
+({									\
+	asm volatile("mfence":::"memory");				\
+	ktsan_membar_acq_rel();						\
```

Comment:
these then should be called:

ktsan_thread_fence(ktsan_memory_order_acquire);
ktsan_thread_fence(ktsan_memory_order_release);
ktsan_thread_fence(ktsan_memory_order_acq_rel);